src/Configuration/Configuration.php

```diff
@@ -7,17 +7,32 @@
 
 namespace Deployer\Configuration;
 
-use Deployer\Exception\ConfigurationException;
 use Deployer\Utility\Httpie;
 use function Deployer\get;
+use Deployer\Exception\ConfigurationException;
 use function Deployer\Support\array_merge_alternate;
 use function Deployer\Support\is_closure;
 use function Deployer\Support\normalize_line_endings;
 
 class Configuration implements \ArrayAccess
 {
+
+    const PREDEFINED_PARAMS = [
```

Comment:
Oh, sure! I forgot there are another ones. At the moment, I cannot really think about any automated way into doing it. Is it ok to keep including them manually at this PREDEFINED_PARAMS array?

Reply:
We can check such an existing on `set()` function call. If there is already something similar to what we are trying to configure - it's a possible error other there.

Reply:
This is a good idea! But we would still need a `PREDEFINED_PARAMS` array to avoid warnings when first setting `bin/dir` and `bin/git`, for example. Actually, it could be triggered every time if any recipe try to use a very similar name as an existing one.

Reply:
If there no previous values, no need to warning. So no need to keep PREDEFINED_PARAMS.

Reply:
Sorry, not sure if I am understanding correctly. Because the wrong configuration would not have a previous value, so it would not trigger the notification as well.

Reply:
There is no such thing wrong configuration before first config are set.

Reply:
Only next set calls can be wrong.

Reply:
Oh, so there a way to identify which `set` calls came from the configuration file! Ok, I will try to find that.

Reply:
I've included a check for which file is calling the `set` function, so the verification is only going to run if it came from the deploy.php file. Could you take a look if that is ok? Thanks! :)